components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DataServiceDocLitWrappedSchemaGenerator.java

```diff
@@ -255,9 +257,17 @@ private static void processRequestInput(CommonParams cparams, CallableRequest re
                                 }
                                 tmpEl = createInputEntryElement(cparams, query, queryParam,
                                         tmpWithParam);
+                                /* checking if query is SQL update query and for optional parameters*/
+                                if (callQuery.getQuery() instanceof SQLQuery
+                                   && ((SQLQuery) query).getSqlQueryType() == SQLQuery.QueryType.UPDATE
+                                   && queryParam.isOptional()) {
+                                   optional = true;
+                                } else {
+                                   optional = false;

```

Comment:
We can replace if - else here and simply assign outcome of the expression to the boolean

Reply:
Fixed

Reply:
fixed